vmdb/app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -99,4 +99,35 @@ def store_ids_for_new_records(records, hashes, keys)
       h[:id] = r.id
     end
   end
+
+  # most of the refresh_inventory_multi calls follow the same pattern
+  # this pulls it out
+  def save_inventory_assoc(type, parent, hashes, target, find_key, child_keys = [], extra_keys = [])
+    reflection = parent.class.reflect_on_association(type)
+    klass = reflection.class_name.constantize
+    deletes = relation_values(parent, type, target)
+
+    save_inventory_multi(type, klass, parent, hashes, deletes, find_key, child_keys, extra_keys)
+    store_ids_for_new_records(parent.send(type), hashes, find_key)
+  end
+
+  # We need to determine our intent:
+  # - make a complete refresh. Delete missing records.
+  # - make a partial refresh. Don't delete missing records.
+  # This generates the "deletes" values based upon this intent
+  # It will delete missing records if both of the following is true:
+  # - The association is declared as a top_level association
+  #   In Active Record, :dependent => :destory says the parent controls the lifespan of the children
```

Comment:
Looks like @kbrock borrowed my keyboard to type that one.  I do that one ALL THE TIME.